### ISSUE TITLE
[BUG]1.2.0 MKS WiFi Delete file fails

### ISSUE THREAD

**Issue by bellzw** [Jeredian/mks-wifi-plugin#115]
**Describe the bug**
When I display files on the printer's SD card, select one, and click Delete, the file is not deleted.

**To Reproduce**
1. Start Cura
2. Connect to printer
3. Click Monitor
4. Click SD Files
5. Select a file (file is highlighted in blue)
6. Click Delete (Open file(s) window disappears)
7. Previously selected file is still present

**Expected behavior**
The selected file should be removed from the SD card.  It would also have been nice if the Open file(s) window did not disappear and simply refreshed.

**Screenshots**
I don't know that screen shots will show anything useful.

**Desktop (please complete the following information):**
 - OS: 64 bit Windows 10 Home Edition running on a Lenovo Yoga 14 laptop
 - Cura version: 4.8.0
 - Plugin version: 1.2.0
 - 3D printer: BIBO2 Dual
 - Motherboard and/or WiFi module firmware version: See photo below
 
![0209211153](https://user-images.githubusercontent.com/51066449/107428772-1bbd8580-6af1-11eb-81e9-2ab53271053f.jpg)

**Additional context**
I have since found that the problem seems to be related to Abort Print. I have just found that if I abort a print started from a file on the SD card, and then click SD Files->Delete, the file is not deleted.  I am pretty sure I had aborted at least one print job before noticing the Delete failed and first posting this.  I apologize for not being sufficiently thorough.

**Comment by Elkin-Vasily** [comment]
Is this issue actual with the latest plugin version?

**Comment by bellzw** [comment]
It was with 1.2.0 installed in Cura 4.8.0.  I have since updated Cura to 4.9.0 and MKS to 1.2.2 and see problems.  I was a little more systematic with my exploration and have attached a Cura log file.

Below is what Cura 4.9.0 shows me when I click SD Files after connecting to the printer.  I have selected file Copy.gcode.
![Before delete](https://user-images.githubusercontent.com/51066449/116010648-51511580-a5ee-11eb-8046-44a85d8bbe53.png)

If I do not select a file, the Delete button does nothing, which is reasonable, although a warning to the effect that the user didn't select a file to delete would be appropriate.

After selecting Copy.gcode,  I click Delete, and then SD Files, I am shows this:
![After delete](https://user-images.githubusercontent.com/51066449/116010701-9f661900-a5ee-11eb-8356-464c72241194.png)

Note that TWO files have disappeared.  However, if I wait 5 - 15 seconds and click SD Files again, I see
![After Marketplace](https://user-images.githubusercontent.com/51066449/116010764-20bdab80-a5ef-11eb-842c-049302e94c8a.png)

The Open file(s) dialog box never seems to show all the files if I leave the dialog open and just keep clicking Refresh.  I seem to have to close the dialog box, and then reopen it with SD Files before I will see all 3 files.

After all this, I took the SD card out of the printer and looked at what is in the root folder:
![SD directory Win10](https://user-images.githubusercontent.com/51066449/116011022-adb53480-a5f0-11eb-97e5-fd05283f6277.png)
I have highlighted the 3 files shown to me by SD Files.

I have attached a Cura log generated only from starting Cura 4.9.0, connecting to my printer, clicking SD Files a few times, selecting and trying to delete Copy.gcode, and then clicking SD Files a few times until the Open File(s) window displayed the 3 gcode files in the root directory.  Perhaps it will help you.
[cura.log](https://github.com/Jeredian/mks-wifi-plugin/files/6373381/cura.log)

I noticed a number of M105 M997 commands sent to the printer.  Isn't M997 a command for updating firmware?  Why would you need to do that every 2 seconds?

At line 432, you will see an M30 command with the file path/name being 1:/Copy.gcode, followed by multiple M20 commands.  The returned file lists still have Copy.gcode in them.  Assuming 1:/ is the correct device, does this indicate that my printer's firmware does not implement file manipulation?  I'll be happy to check with the manufacturer (BIBO).

Bottom line: files are still not actually deleted and the plug-in does not display the complete contents of the SD card's root folder, even though it gets the complete list of files in response to an M20.
